engine/evd.ml

```diff
@@ -642,6 +643,7 @@ type evar_map = {
   defn_evars : defined evar_info EvMap.t;
   undf_evars : undefined evar_info EvMap.t;
   evar_names : EvNames.t;
+  candidate_evars : Evar.Set.t; (* The subset of undefined evars with a non-empty candidate list. *)
```

Comment:
why not a map to the candidate set, moving that data away from evar_info?

Reply:
I'd rather have potentially memory-heavy data in the evarinfo, this way it's clear it gets erased when defined.